### PR TITLE
Clean up MDAs from Interop code

### PR DESCRIPTION
**Pull request by AaronRobinsonMSFT**

### CODE REVIEW

**Comment by AaronRobinsonMSFT** [comment]
Related https://github.com/dotnet/coreclr/issues/22538

File: src/vm/dllimport.cpp [hunk 1]
```diff
@@ -2217,28 +2138,8 @@ void NDirectStubLinker::Begin(DWORD dwStubFlags)
...
-            EmitCallGcCollectForMDA(m_pcsSetup, dwStubFlags);
-        }
-#endif // MDA_SUPPORTED
-
         if (SF_IsDelegateStub(dwStubFlags))
         {
-#if defined(MDA_SUPPORTED)
-            // GC was induced (gcUnmanagedToManagedMDA), arguments have been marshaled, and we are about
-            // to touch the UMEntryThunk and extract the delegate target from it so this is the right time
-            // to do the collected delegate MDA check.
```
Comment (stephentoub) [inline src/vm/dllimport.cpp]: Deleting all of this dead code is goodness.  But I do wonder whether some of these checks would be worth keeping around in a different form, as some of them can help to find real problems.  As MDAs, they were always really hard to enable, and very few people did, but are there any such checks that we could do with little-to-no overhead on the success path and either throw or fail fast or raise an EventSource/ETW event or some such thing on the failure path?  Using a collected delegate seems like a good example, though maybe 9/10 times it'll crash immediately so we don't need to do anything special?  PInvokeStackImbalance is one that I know has saved me a few times in the past, too, and that one's harder to detect and often just results in corruption found later that can be difficult to trace back... that one though would seem to add overhead even in the success case, so maybe we'd need some kind of opt-in mechanism for that, a COMPlus_ flag or something.

Comment (AaronRobinsonMSFT) [reply src/vm/dllimport.cpp]: >some of these checks would be worth keeping around in a different form

They definitely are. I wanted to use this PR as a canary to ensure no tests were relying on them. Several should be converted into asserts, process terminate or even perhaps a normal runtime exception (e.g. detected stack imbalance which is always a mismatched calling convention). I am also considering an EventSource/ETW logging mechanism. Definitely will enrich the PR as I discover mechanisms that satisfy key MDA scenarios.

**Comment by AaronRobinsonMSFT** [comment]
Closing in lieu of https://github.com/dotnet/coreclr/pull/24457

**Pull request closed without merging**